addons/mrp/tests/test_consume_tracked_component.py

```diff
@@ -0,0 +1,161 @@
+
+        quant = self.create_quant(self.raw_none, 3*all_qty)
+        quant |= self.create_quant(self.raw_lot, 2*all_qty)
+        quant |= self.create_quant(self.raw_serial, 1*all_qty)
+        quant.action_apply_inventory()
+
+        #Quantities are fully reserved (stock.move state is available)
+        mo_all.action_assign()
+        for mov in mo_all.move_raw_ids:
+            self.assertEqual(mov.product_qty, mov.reserved_availability, "Reserved quantity shall be equal to To Consume quantity.")
+
+        self.autoTriggers(mo_serial)
+        self.autoTriggers(mo_none)
+        self.autoTriggers(mo_lot)
+        for mov in mo_all.move_raw_ids:
+            self.assertEqual(mov.product_qty, mov.quantity_done, "Done quantity shall be equal to To Consume quantity.")
+
+    def test_option_enabled_and_qty_not_available(self):
+        """
+        Option enabled, qty not available
+        -> Not Tracked components are fully consumed
+        -> Tracked components are not consumed
+        """
+
+        mo_none = self.create_mo(self.mo_none_tmpl, self.DEFAULT_TRIGGERS_COUNT)
+        mo_serial = self.create_mo(self.mo_serial_tmpl, self.SERIAL_TRIGGERS_COUNT)
+        mo_lot = self.create_mo(self.mo_lot_tmpl, self.DEFAULT_TRIGGERS_COUNT)
+
+        mo_all = mo_none + mo_serial + mo_lot
+        mo_all.action_confirm()
+
+        #Quantities are not reserved at all (stock.move state is confirmed)
+        mo_all.action_assign()
+        for mov in mo_all.move_raw_ids:
+            self.assertEqual(0, mov.reserved_availability, "Reserved quantity shall be equal to 0.")
+
+        self.autoTriggers(mo_serial)
+        self.autoTriggers(mo_none)
+        self.autoTriggers(mo_lot)
+
+        for mov in mo_all.move_raw_ids:
+            if mov.has_tracking == 'none':
+                self.assertEqual(mov.product_qty, mov.quantity_done, "Done quantity shall be equal to To Consume quantity.")
+            else:
+                self.assertEqual(0, mov.quantity_done, "Done quantity shall be equal to To Consume quantity.")
+
+    def test_option_enabled_and_qty_partially_available(self):
+        """
+        Option enabled, qty partially available
+        -> Not Tracked components are fully consumed
+        -> Tracked components are partially consumed
+        """
+
+        #update BoM serial component qty
+        self.bom_none_lines[2].product_qty = 2
+        self.bom_serial_lines[2].product_qty = 2
+        self.bom_lot_lines[2].product_qty = 2
+
+        raw_none_qty = 2
+        raw_tracked_qty = 1
+
+        quant = self.create_quant(self.raw_none, raw_none_qty)
+        quant |= self.create_quant(self.raw_lot, raw_tracked_qty)
+        quant |= self.create_quant(self.raw_serial, raw_tracked_qty)
+        quant.action_apply_inventory()
+
+        #We must create & process each MO at once as we must assign quants for each individually
+        def testUnit(mo_tmpl, serial=None):
```

Comment:
I'm not exactly sure what `serial` represents? Is it which serial trigger it is testing?